.github/workflows/continuous-integration-workflow.yml

```diff
@@ -87,98 +86,98 @@ jobs:
-        python3 -m venv venv-build
-        ./venv-build/bin/pip install --upgrade pip build twine
-        ./venv-build/bin/python -m build
-        ./venv-build/bin/python -m twine check dist/*
-
-    # TEST_PYPI_API_TOKEN is an API token created on
-    # https://test.pypi.org/manage/account/#api-tokens with restriction to speculos project
-    - name: Publish Python package to test.pypi.org
-      if: success() && github.event_name == 'push' && github.ref == 'refs/heads/master'
-      run: |
-        ./venv-build/bin/python -m twine upload --repository testpypi dist/*
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.TEST_PYPI_API_TOKEN }}
-        TWINE_NON_INTERACTIVE: 1
+      - name: Clone
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Use pip to install Speculos in a virtual environment
+        run: |
+          python3 -m venv venv-test
+          ./venv-test/bin/pip install .
+          ./venv-test/bin/speculos --help
+
+      # Use commands from https://packaging.python.org/tutorials/packaging-projects/
+      # to build the Speculos package, but using a dedicated virtual environment
+      # to be cleanly upgrade Python packages.
+      - name: Build Speculos python package
+        run: |
+          if [ -e dist ] ; then
+            echo >&2 "Error: dist/ directory already exists and this is unexpected. Refusing to build new packages."
+            exit 1
+          fi
+          sed -i 's;docs/screenshot-api-nanos-btc\.png;https://raw.githubusercontent.com/LedgerHQ/speculos/master/docs/screenshot-api-nanos-btc.png;' README.md
+          ./tools/update_setup_version_from_git.sh
+          python3 -m venv venv-build
+          ./venv-build/bin/pip install --upgrade pip build twine
+          ./venv-build/bin/python -m build
+          ./venv-build/bin/python -m twine check dist/*
+
+      # TEST_PYPI_API_TOKEN is an API token created on
+      # https://test.pypi.org/manage/account/#api-tokens with restriction to speculos project
+      - name: Publish Python package to test.pypi.org
+        if: success() && github.event_name == 'push' && github.ref == 'refs/heads/master'
+        run: |
+          ./venv-build/bin/python -m twine upload --repository testpypi dist/*
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          TWINE_NON_INTERACTIVE: 1
 
   package_and_test_docker:
     name: Build and test the Speculos docker
     runs-on: ubuntu-latest
     container:
       image: docker://ghcr.io/ledgerhq/speculos-builder:latest
     steps:
-    - name: Clone
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
-    - name: Set up Docker Buildx
+      - name: Clone
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Docker Buildx
         id: buildx
         uses: docker/setup-buildx-action@master
```

Comment:
Please use a specific version instead of `@master` everywhere (`@v2` seems to be the latest release).

Reply:
done